### DISCUSSION TITLE
fusing GPS and Optical Flow data

### DISCUSSION THREAD

**Discussion by aistaca** [iNavFlight/inav#8007] [Q&A] [answered]
Hello,
I'm new to INAV. Does INAV support fusing GPS and Optical Flow data? If I have both GPS and optical flow, will the opflow help increase position accuracy, or if there's no satellite (or low HDOP)? Let's say I have both sensors enabled and flying indoor, will it fly in poshold using opflow, and switch to gps once it get gps lock?

**Comment by DzikuVx** [comment, marked as answer]
Those are different altitudes and are used in different flight modes. They are not fused as they measure different things. GPS/Baro measures relative altitude comparing to home point. LIDAR measures current over the ground altitude. They are independent

**Reply by aistaca** [reply]
I see, thank you for the answer.

I was hoping for the optical flow to give more horizontal position accuracy in tandem with GPS, like what they did in ardupilot.

**Reply by DzikuVx** [reply]
The thing with all those available sensors is that they all have very limited range
